### PR TITLE
Collect cloud-init logs

### LINKED ISSUE #30411: node e2e - collect cloud init logs
If node e2es fail, it is not possible to figure out if node initialization failed. We need to collect cloud init logs as well.

### PR DESCRIPTION
**Pull request by vishh**
Fixes #30411

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/30467)

### CODE REVIEW

File: test/e2e_node/e2e_service.go [hunk 1]
```diff
@@ -66,8 +66,9 @@ const (
 func newE2eService(nodeName string, cgroupsPerQOS bool, evictionHard string, context *SharedContext) *e2eService {
 	// Special log files that need to be collected for additional debugging.
 	var logFiles = map[string]logFileData{
-		"kern.log":   {[]string{"/var/log/kern.log"}, []string{"-k"}},
-		"docker.log": {[]string{"/var/log/docker.log", "/var/log/upstart/docker.log"}, []string{"-u", "docker"}},
+		"kern.log":       {[]string{"/var/log/kern.log"}, []string{"-k"}},
+		"docker.log":     {[]string{"/var/log/docker.log", "/var/log/upstart/docker.log"}, []string{"-u", "docker"}},
+		"cloud-init.log": {[]string{}, []string{"-u", "cloud*"}},
```
Comment (timstclair) [inline test/e2e_node/e2e_service.go]: I see `/var/log/cloud-init.log` on my ubuntu instance. Add it to the file list?

**Comment by timstclair** [comment]
One small comment, you can self-apply LGTM once you address it.

File: test/e2e_node/e2e_service.go (hunk 1 above)
Comment (vishh) [inline test/e2e_node/e2e_service.go]: Are there any other log files related to cloud init? Like `cloud-config` for example?

**Comment by vishh** [comment]
Added a regular log file for cloud init. Waiting on jenkins results before adding lgtm label.

**Comment by vishh** [comment]
Verified that log data is being exported. [logs from GCI](https://00e9e64bac00b06b6db5c4314199e0d1dd1503834d81c5ab49-apidata.googleusercontent.com/download/storage/v1/b/kubernetes-jenkins/o/pr-logs%2Fpull%2F30467%2Fnode-pull-build-e2e-test%2F18393%2Fartifacts%2Ftmp-node-e2e-f3820b23-gci-dev-54-8666-0-0%2Fcloud-init.log?qk=AD5uMEvnsCC8U-cOJ_nPTzCLLSue0aBArpjtdoALSGEpn9lJty1II6Gwv-kGCMFWqFm0zvdDLBO_jZTTXnGlfo4fuZUbUm44IwEDkQc8hP9RV1hNeKp2mU6NYBzYMqfzBwn3uJlvdqy0R9EQPyaBpyX1KYN7e13DLkyFJy526LKTlHy3n_qfq5H86CfZJKHDKD-wXEhsfRwqbCUOdLgMQxL9UidSrp7TUSEjuIRQFcBGGU5QjLlLuNtQV5NumhghXS28z8rd05N5Z7GNDwFIdZSxxnWNB9OetqR-DNS0uKzJ_uv6gZrUbpLLYTBWxXQf7AKY4r705fxSSqvUzTij7akgD30nRuFrky0AJJByuQZP2ovjOoegji3erxDaxG7tSIlKSkSYjmR6872vSviiCOIMnbHBXa__haFRRcFxN1-Ueki0jUPAsQBKfPB6H0ujLu5nbHVk_5mRzAaLt-OmQ0cXIfOSFr1HDsHX76R6k0JDXvvHy2ydGfBghYkWbSmf7S_72tttrVJK4g6m47p1-CivuBFqmlclrkStVLBGlNsXWT932Bcf72K3nfQhxhN0w0ktNilFAbTPQPDf_1wFVZxNJ2WReF5GqFG_bIsdjZrgu1e3_zmT3vfSk9vz6pz2qPkZZGBHWgtFrhuXwkumvDZwr5PBlI3dn9n4Zam6EOPbIILKQBcgjp7mFnclC4F5Iy3bnri8uWijVnsDclpKpzCzCkoWWStwa4KN4Yeey1fO1KzSbU4CMDCOpSss7-l4Fwa7kIJ6y1i8mgYjAakfGrlkuCWujGqOre-sG-T8sMLnCLibapB580F8yTPL25LIJBB3CqpgAzPwV-O6G8dFiyQiIkmd3-bhhWT4PdJK273fIz0VDEMeNYdrt3srw02po9ovJfgInKbrtOGzwQLOE7sY3Tv72-v5WpLA3ITfqBW0lVdW_fslqdg)

**Pull request merged**